linters/trufflehog/trufflehog_to_sarif.py

```diff
@@ -48,11 +75,11 @@ def main(argv):
             description = "Secret detected"
 
         if "Filesystem" in vuln_json["SourceMetadata"]["Data"]:
+            secret = vuln_json["Raw"]
             path = vuln_json["SourceMetadata"]["Data"]["Filesystem"]["file"]
-            line_number = (
-                int(vuln_json["SourceMetadata"]["Data"]["Filesystem"].get("line", "0"))
-                + 1
-            )
+            line_number = find_line_number(secret, path)
+            if line_number is None:
+                continue
```

Comment:
ugh... i'm kind of torn on this behavior. i don't know if we want to fallback to the trufflehog-specified line number or line 0.

discarding the diagnostic feels wrong though- i don't think we've ever seen trufflehog just completely hallucinate an issue, it's always just wrong location, right?

Reply:
Trufflehog can report the same issue multiple times - for instance, the test file I added reports 4 secrets, reporting each of the secrets twice.

Reply:
gah, ffs

let's add this as an inline comment here as well (i know you mentioned it above)